Raise a ResponseError for FedEx#find_tracking_info

thegedge opened the pull request:
In a couple of places we were raising an `ActiveShipping::Error` when we could raise/return a more informed `ActiveShipping::TrackingResponse(false, ...)` that has additional information (e..g, response, request) for clients of this gem.

The one point of discussion here is w.r.t. the semantics of `ResponseError`. Generally it means there was an issue in the response, but for these two cases I've changed aren't considered error responses by FedEx, and aren't given special status codes (i.e., `/TrackReply/Notifications/Code` is 0), so perhaps this doesn't make sense?

Also made a few small formatting changes on nearby code (separate commit).

jonathankwok commented on test/unit/carriers/fedex_test.rb:
```diff
@@ -327,6 +327,23 @@ def test_failure_to_parse_invalid_xml_results_in_a_useful_error
+      location_fixtures[:ottawa],
+      location_fixtures[:beverly_hills],
+      package_fixtures.values_at(:book, :wii),
+      :test => true
```
`test: true` would be 👍  here.

jonathankwok commented on test/unit/carriers/fedex_test.rb:
```diff
@@ -327,6 +327,23 @@ def test_failure_to_parse_invalid_xml_results_in_a_useful_error
+      :test => true
+    )
+
+    assert response.success?
```
`assert_predicate`

jonathankwok commented on test/unit/carriers/fedex_test.rb:
```diff
@@ -507,14 +507,25 @@ def test_tracking_info_without_status
     assert_equal msg, error.message
   end
 
+  def test_tracking_info_without_tracking_details
```
```
test "#find_tracking_info without tracking details" do
```

jonathankwok replied:
You could also change it throughout this PR. Or, keep it as is and have a dedicated PR to updating this test file to `test "" do`

thegedge replied:
Yeah, I'd rather keep that as a bigger cleanup PR to keep things cohesive.

thegedge commented:
Updated

thegedge commented:
Okay, I added one more commit which was the actual real issue that needed to be dealt with here, but we can keep the `TrackingResponse` code here too.

tl;dr we were only looking at the success status from the document root, and in a `TrackDetails` element, but forgetting to check `CompleteTrackDetails`. I've updated the test to capture this case.

thegedge commented:
^ Squashed it into two, cohesive commits that I'll merge (they don't make sense to merge together)
